components/ProfileDropdownMenu.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import Dropdown from 'react-bootstrap/Dropdown'
+import UserInfoImage from './UserInfoImage'
+import { ChevronDownIcon } from '@primer/octicons-react'
+import LogoutContainer from './LogoutContainer'
+import { Button } from './theme/Button'
+import { UserInfo } from '../@types/user'
+import { useUserInfoQuery } from '../graphql/index'
+import { useRouter } from 'next/router'
+import _ from 'lodash'
+import styles from '../scss/profileDropDown.module.scss'
+
+type ProfileDropDownMenuProps = {
+  username: string
+  isAdmin: boolean
+}
+
+type CustomToggleProps = {
+  children?: React.ReactNode
+  className?: string
+  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
+}
+
+const ProfileDropdownMenu: React.FC<ProfileDropDownMenuProps> = ({
+  username,
+  isAdmin
+}) => {
+  const { data } = useUserInfoQuery({
+    variables: { username },
+    skip: !username
+  })
+  const router = useRouter()
+  const location = router.asPath
+  const PROFILE_PATH = '/profile/' + username
```

Comment:
profile path should go in a constants file that should be accessible from any part of the codebase. I see a high potential for all `path` values to be re-used in the codebase. For example `/curriculum` appears a lot
https://github.com/garageScript/c0d3-app/search?q=curriculum

Reply:
Right, although the profile path is only used in the profile dropdown menu for now, I feel like this needs to be another different PR, the constants file doesn't exist yet.
I'll reference this issue here so that we can keep track of it: #1185

Reply:
So far the only file that uses `/profile` is components/AppNav, but now we are introducing a new file components/ProfileDropdownMenu in this PR that uses it as well. We shouldn't be introducing more tech debt. `components/AppNav` is already an impacted file, so you're really only adding 1 more file to your PR. This current PR is kinda big already though, so I'd suggest to split this PR into smaller ones.

Reply:
Created Constants for all the paths in AppNav and ProfileDropdownMenu.